packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,200 @@
+
+type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to describe the popover content. Used for [accessibility](https://gestalt.pinterest.systems/web/popover#ARIA-attributes) purposes.
+   */
+  accessibilityPopoverLabel: string,
+  /**
+   * Specifies the preferred position of the tooltip and popover relative to HelpButton. See [Popover's ideal direction variant](https://gestalt.pinterest.systems/web/popover#Ideal-direction) to learn more.
+   */
+  idealDirection?: 'up' | 'right' | 'down' | 'left',
+  /**
+   * If provided, displays a link at the bottom of the popover message.
+   * - `href` is the URL the hyperlink points to.
+   * - `text` is the displayed text for the link. See the [link variant](https://gestalt.pinterest.systems/web/helpbutton#With-a-link) for more details.
+   * - Optionally use `accessibilityLabel` to supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like "Click Here", or "Read More" can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates `aria-label`. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See [ Link's accessibility guidelines](https://gestalt.pinterest.systems/web/link#Accessibility) for more information.
+   * - Optionally provide an `onClick` callback, which is fired when the link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   */
+  link?: {|
+    accessibilityLabel?: string,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    text?: string,
+  |},
+  /**
+   * Callback fired when HelpIcon is clicked (pressed and released) with a mouse or keyboard.
+   */
+  onClick?: OnTapType,
+  /**
+   * Informational content that's displayed when the user clicks on HelpButton.
+   */
+  text: string | Element<typeof Text>,
+  /**
+   * Specifies the z-index for HelpButton's tooltip and popover to resolve any layering issues with other elements. See the [zIndex variant](https://gestalt.pinterest.systems/web/helpbutton#With-Z-index) for more details.
+   */
+  zIndex?: Indexable,
+|};
+
+/**
+ * [HelpButton](https://gestalt.pinterest.systems/web/helpbutton) is an affordance that accompanies an element on the screen. It helps describe or provide assistance on how to use the accompanying element.
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  onClick,
+  text,
+  zIndex,
+}: Props): Node {
+  const ref = useRef(null);
```

Comment:
I'd rename this `tapAreaRef` to know what's targeting. Makes more sense on line 78 to be `tapAreaRef`